### PR TITLE
Consume library version of force:apex:log:get in VS Code

### PR DESCRIPTION
**Pull request by smitshah2531**
### What does this PR do?

With this PR, we are now using the Apex library instead of the CLI to run the `force:apex:log:get` command

### What issues does this PR fix or reference?
 @W-7561093@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json [hunk 1]
```diff
@@ -803,6 +803,11 @@
           "type": "boolean",
           "default": false,
           "description": "%experimental_deploy_retrieve_description%"
+        },
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/package.json:806]: You shouldn't need to add the flag bc it was already added through a different PR. If you rebase from develop, you should see the changes!

File: packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts [hunk 2]
```diff
@@ -208,14 +214,95 @@ export class ForceApexLogList {
   }
 }
 
-const workspaceChecker = new SfdxWorkspaceChecker();
-const parameterGatherer = new LogFileSelector();
+export class LogGetGatherer implements ParametersGatherer<{ Id: string }> {
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts:217]: re: earlier conversation, we should be able to use `LogFileSelector` for both implementations

File: packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts [hunk 3]
```diff
@@ -81,4 +81,33 @@ export abstract class ApexLibraryExecutor extends LibraryCommandletExecutor<{
...
+        }
+      );
+
+      channelService.showCommandWithTimestamp(`Finished ${commandName}`);
+
+      const logPath = path.join(`${args[0].outputDir}`, `${args[0].logId}.txt`);
+      const document = await vscode.workspace.openTextDocument(logPath);
+      vscode.window.showTextDocument(document);
+
+      // Notification successful message
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts:108]: Add code for showing successful message

File: packages/salesforcedx-vscode-core/package.json [hunk 4]
```diff
@@ -803,6 +803,11 @@
...
+        "salesforcedx-vscode-core.experimental.useApexLibrary": {
+          "type": "boolean",
+          "default": true,
+          "description": "%experimental_apex_library_description%"
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/package.json:810]: We should update the `apex-node` dependency to `0.0.3` so we have access to the latest changes in the library

File: packages/salesforcedx-vscode-core/package.json [hunk 5]
```diff
@@ -798,11 +798,6 @@
           "default": false,
           "description": "%conflict_detection_enabled_description%"
         },
-        "salesforcedx-vscode-core.experimental.deployRetrieve": {
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/package.json:801]: We still need the `deployRetrieve` setting, let's not touch that :)

Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/package.json:801]: Sounds good!

**Comment by AnanyaJha** [comment]
I've been testing this and it looks like the cli command typically expects a `.log` file to be generated. In the Apex library, we're currently outputting the log to a `.txt` file. I think it'd be worthwhile for us to fix that in the apex library and update the file name accordingly here.

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts [hunk 6]
```diff
@@ -209,6 +209,8 @@ export const messages = {
   force_apex_log_get_no_logs_text: 'No Apex debug logs were found',
   force_apex_log_get_pick_log_text: 'Pick an Apex debug log to get',
   force_apex_log_list_text: 'Getting Apex debug logs',
+  force_apex_log_get_library: 'Apex Library: Get Logs',
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: @hbc-bindu @ekapner What do we think about the message "Log Get Apex". This text will be displayed in the progress bar while the command is running: 
Screen Shot 2020-07-17 at 12 41 38 PM

Comment (AnanyaJha) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: We currently display the text used to run the CLI command behind the scenes:
Screen Shot 2020-07-17 at 12 42 19 PM

File: packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts [hunk 7]
```diff
@@ -208,14 +214,60 @@ export class ForceApexLogList {
...
+    ? new ApexLibraryGetLogsExecutor()
+    : new ForceApexLogGetExecutor();
+
   const commandlet = new SfdxCommandlet(
-    workspaceChecker,
-    parameterGatherer,
-    new ForceApexLogGetExecutor()
+    new SfdxWorkspaceChecker(),
+    parameterGatherer as ParametersGatherer<{}>,
+    logGetExecutor as CommandletExecutor<{}>
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts:270]:
```suggestion
    parameterGatherer,
    logGetExecutor
```

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts (hunk 6 above)
Comment (ekapner) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: @AnanyaJha - what about? And is it logs, plural, or just ONE log? I noticed the command above says "Get Logs"
Starting Get Apex Log
Finished Get Apex Log 
If could be multiple logs, then I would just add an "s" to Log

File: packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts:270]: 👍🏻

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts (hunk 6 above)
Comment (AnanyaJha) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: Sounds good to me, we can go with `Get Apex Logs`, thanks!

Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: Yeah I was thinking to change the name as well. Like "Log Get Apex" sounds little bit weird so lets go with "Get Apex Logs" 👍🏻

Comment (ekapner) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: Thank you!

Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: No customer facing message should have a reference to a change in the implementation (library vs cli). This should be changed to `Get Apex Logs` or `SFDX: Get Apex Logs`.

Comment (AnanyaJha) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: Yeah this message is only used for telemetry, the customer facing message is `Get Apex Logs`

Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: Yeah customer facing message is "apex_log_get_text: 'Get Apex Logs' " !

Comment (ekapner) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: @smitshah2531 and @AnanyaJha - do you have everything you need from me?

Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: @ekapner I guess yes. This was the only one thing I guess we were worried about. From my side everything looks good to me. Do you have anything @AnanyaJha ?

Comment (AnanyaJha) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:212]: Yes, that was the only customer facing text. thanks for the review emily!

File: packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts [hunk 8]
```diff
@@ -81,4 +81,31 @@ export abstract class ApexLibraryExecutor extends LibraryCommandletExecutor<{
...
+        },
+        async () => {
+          // @ts-ignore
+          return (await fn.call(this, ...args)) as string[];
+        }
+      );
+
+      channelService.showCommandWithTimestamp(`Finished ${commandName}`);
+
+      const logPath = path.join(`${args[0].outputDir}`, `${args[0].logId}.txt`);
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts:104]: What is the content of this file, the debug log ? If so, it needs to be updated to `.log`

File: packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts [hunk 9]
```diff
@@ -208,14 +214,60 @@ export class ForceApexLogList {
...
+      await this.logService.getLogs({
+        logId: id,
+        outputDir: logDir
+      });
+    } catch (e) {
+      telemetryService.sendException(
+        nls.localize('force_apex_log_get_library'),
+        e.message
+      );
+      notificationService.showFailedExecution(this.executionName!);
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts:255]: Nit: we don't need the `!` 
```suggestion
      notificationService.showFailedExecution(this.executionName);
```

File: packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts [hunk 10]
```diff
@@ -81,4 +81,31 @@ export abstract class ApexLibraryExecutor extends LibraryCommandletExecutor<{
...
+        },
+        async () => {
+          // @ts-ignore
+          return (await fn.call(this, ...args)) as string[];
+        }
+      );
+
+      channelService.showCommandWithTimestamp(`Finished ${commandName}`);
+
+      const logPath = path.join(`${args[0].outputDir}`, `${args[0].logId}.log`);
```
Comment (lcampos) [inline packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts:104]: We're not returning the file path for the log that got retrieved as part of the library's response ?

Comment (AnanyaJha) [reply packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts:104]: No I believe we're returning the logs themselves as the response

Comment (lcampos) [reply packages/salesforcedx-vscode-core/src/commands/util/apexLibraryExecutor.ts:104]: Got it, we should take a look at the response and have it return everything we might need to consume so we don't have to do this on the consumer side. Let's log something for that in the library.

**Pull request merged**